FlowCrypt/Controllers/Inbox/InboxViewController.swift

```diff
@@ -452,10 +452,22 @@ extension InboxViewController: MsgListViewController {
                 tableNode.reloadData()
             } else {
                 state = .fetched(.byNumber(total: newTotalNumber))
-                tableNode.deleteRows(at: [IndexPath(row: index, section: 0)], with: .left)
+                do {
+                    try ObjcException.catch {
```

Comment:
Can you please explain how is this fix related to the crash fix?

Reply:
It looks like there is an issue related to the index of the row that we try to update

Reply:
As @tomholub said, we want to avoid having crash here. Even if issue is still here, we should not crash.

Reply:
Can we have an issue to solve the root cause of the crash?

Reply:
absolutely. Let's solve it in this PR as well.

Reply:
Since all UI is async with Texture, I guess that using message index is unsafe. We should move to message ID, as this is the only way to know that message we want to delete is the same message.

Reply:
Would suggest to investigate root cause of the issue. 

Also if there was only one message before moving to trash we need to update whole state of the controller.